pkg/sfu/downtrack.go

```diff
@@ -895,17 +889,20 @@ func (d *DownTrack) getTranslatedRTPHeader(extPkt *buffer.ExtPacket, tpRTP *Tran
 	return &hdr, nil
 }
 
-func (d *DownTrack) translateVP8Packet(pkt *rtp.Packet, incomingVP8 *buffer.VP8, translatedVP8 *buffer.VP8) (buf []byte, err error) {
-	buf = *d.payload
-	buf = buf[:len(pkt.Payload)+translatedVP8.HeaderSize-incomingVP8.HeaderSize]
+func (d *DownTrack) translateVP8Packet(pkt *rtp.Packet, incomingVP8 *buffer.VP8, translatedVP8 *buffer.VP8) (buf []byte, pool *[]byte, err error) {
+	if incomingVP8.HeaderSize == translatedVP8.HeaderSize {
+		buf = pkt.Payload
+	} else {
+		pool = PacketFactory.Get().(*[]byte)
```

Comment:
what about pass buf as parameter? split put and get to different function is not straightforward and easy to make error

Reply:
Good point @cnderrauber . I thought about it, but got a bit lazy🙂. Also, there were too many return paths. Will have another look.